src/cargo/core/gc.rs

```diff
@@ -0,0 +1,550 @@
+//! Support for garbage collecting unused files from downloaded files or
```

Comment:
A part of me feels like this doesn't belong in `core` but I don't have a big enough picture yet to say for sure or where this should go but leaving this comment in case it inspires thoughts in others.